### PR TITLE
refactor: Add registry scanner module

### PR DESCRIPTION
**Pull request by ishitasequeira**
Image Updater provides some functionality that can be reused by other projects, most notably the feature to inspect OCI and Docker registries' contents and pick an image based on some constraints.

This PR starts the initial setup of a new module named `registry-scanner` to separate out the functionality of registry scanning and fetching images into a library. This allows users to pull in just the registry scanning functionality and not pull in the entire image updater for use.

Future Tasks:
* Replicate part of `pkg/` folders to registry-scanner baring the argo-cd functionality
* Integrate `registry-scanner` with argocd-image-updater
* Update github CI to execute registry-scanner checks only when updating files in the module
* Releasing `registry-scanner` as part of release process.

### CODE REVIEW

File: registry-scanner/Makefile [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+.PHONY: lint
+lint:
+	golangci-lint run
+
+.PHONY: registry-scanner-image
+registry-scanner-image:
+	docker build -t $(IMAGE_PREFIX)registry-scanner . -f Dockerfile
+	@if [ "$(DOCKER_PUSH)" = "true" ] ; then docker push $(IMAGE_PREFIX)registry-scanner:$(IMAGE_TAG) ; fi
+
```
Comment (jannfis) [inline registry-scanner/Makefile:37]: Not sure if we'll ever need to publish an image of this. I'd skip that part.

File: registry-scanner/README.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+Registry Scanner is under active development. We would not recommend it
+yet for *critical* production workloads, but feel free to give it a spin.
+
+We're very interested in feedback on usability and the user experience as well
+as in bug discoveries and enhancement requests.
+
+**Important note:** Until the first stable version (i.e. `v1.0`) is released,
+breaking changes between the releases must be expected. We will do our best
+to indicate all breaking changes (and how to un-break them) in the
+[Changelog](CHANGELOG.md)
```
Comment (jannfis) [inline registry-scanner/README.md:21]: I'd replace this by a statement that this component is not yet ready for public consumption, and that we're currently working on a first release.

File: registry-scanner/LICENSE [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "[]"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright [yyyy] [name of copyright owner]
```
Comment (jannfis) [inline registry-scanner/LICENSE:189]:
```suggestion
   Copyright 2024 The Argo CD Image Updater authors
```

Comment (ishitasequeira) [reply registry-scanner/LICENSE:189]: I see this to be same in Image Updater.. Should I update it their as well?

**Review by jannfis — APPROVED** [review]

**Pull request merged**